main-app/app/scripts/gameboard/gameboard-services/gameboard-service.js

```diff
@@ -0,0 +1,41 @@
+angular.module('Server.MyModule')
+    .service('FreshGame', ['PlayerService','NewGame','CoreData','ServerTurn','$state',  function (playerService, newGame, coreData, serverTurn, $state) {
+        var outcome = 'Continue',
+            me = this,
+            gameEnded = function () {
```

Comment:
This isn't really part of the game's responsibility. I'd inject an endOfGame service that handles the three end of game conditions. This makes the response pluggable.